### ISSUE TITLE
Error on Sniper Team

### ISSUE THREAD

**Issue by desertking** [ruffle-rs/ruffle#10996]
Dunno if those XForm games will be ever supported by Ruffle but they had a big audience so I post it here.

# Error Info
Error name: Error
Error message: panicked at 'wgpu error: Validation Error

Caused by:
    In Device::create_texture
    Dimension X is zero

', /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/wgpu-0.16.0/src/backend/direct.rs:3019:5
Error stack:
```
Error: panicked at 'wgpu error: Validation Error

Caused by:
    In Device::create_texture
    Dimension X is zero

', /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/wgpu-0.16.0/src/backend/direct.rs:3019:5
    at O.n.wbg.__wbg_new_15d3966e9981a196 (https://www.spiele-umsonst.de/ruffle/core.ruffle.1f63c5b1396ac9e10d6d.js:1:49872)
    at core::ops::function::FnOnce::call_once{{vtable.shim}}::haa731965db7740b6 (wasm://wasm/02c2d3da:wasm-function[8572]:0x830b2a)
    at std::panicking::rust_panic_with_hook::h7f7102b82d51338f (wasm://wasm/02c2d3da:wasm-function[5598]:0x754c53)
    at core::panicking::panic_fmt::he4489d678d6570d5 (wasm://wasm/02c2d3da:wasm-function[8421]:0x82b28e)
    at core::ops::function::Fn::call::h4aa4b0d30ae7a730 (wasm://wasm/02c2d3da:wasm-function[5933]:0x76998a)
    at wgpu::backend::direct::ErrorSinkRaw::handle_error::h647e6d97577ac779 (wasm://wasm/02c2d3da:wasm-function[4418]:0x6f17d4)
    at <T as wgpu::context::DynContext>::device_create_texture::h67a6200fd6e78a48 (wasm://wasm/02c2d3da:wasm-function[6247]:0x786d54)
    at wgpu::Device::create_texture::h841bd1e926825a58 (wasm://wasm/02c2d3da:wasm-function[4333]:0x6e91ee)
    at <ruffle_render_wgpu::context3d::WgpuContext3D as ruffle_render::backend::Context3D>::process_command::{{closure}}::h60764e43d9244631 (wasm://wasm/02c2d3da:wasm-function[2360]:0x54fc26)
    at <ruffle_render_wgpu::context3d::WgpuContext3D as ruffle_render::backend::Context3D>::process_command::h66ae88d8b6f4cb64 (wasm://wasm/02c2d3da:wasm-function[529]:0xc7bb7)
```
AVM2 stack:
```
    at flash.display3D::Context3D/flash::display3D::Context3D::configureBackBuffer()
    at away3d.core.managers::Stage3DProxy/onContext3DUpdate()
    at flash.events::EventDispatcher/flash::events::EventDispatcher::dispatchEvent()
    at flash.display::Stage3D/flash::display::Stage3D::requestContext3D_internal()
    at MethodInfo-1454()
```

# Player Info
Allows script access: false

# Page Info
SWF URL: https://www.spiele-umsonst.de/azad/downloads/flashgames/sniperteam/game.swf
Needed data: https://www.spiele-umsonst.de/azad/downloads/flashgames/sniperteam/Shared.xfg

# Browser Info
User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/113.0.0.0 Safari/537.36
Platform: Win32
Has touch support: false

# Ruffle Info
Version: 0.1.0
Name: nightly 2023-05-09
Channel: nightly
Built: 2023-05-09T00:25:26.314Z
Commit: 72769eeefa66b617d3d7f710a76f4ea78ec16104
Is extension: false

# Metadata
width: 800
height: 450
frameRate: 30
numFrames: 2
swfVersion: 13
backgroundColor: undefined
isActionScript3: true
uncompressedLength: 3982923

**Comment by n0samu** [comment]
This has been fixed, but there's a new panic now:
```
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `1`,
 right: `0`', core\src\pixel_bender.rs:374:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
AVM2 stack trace:
        at flash.display::ShaderData/flash::display::ShaderData::init()
        at flash.display::ShaderData()
        at flash.display::Shader/set byteCode()
        at flash.display::Shader()
        at away3d.core.raycast.colliders::TriangleCollider()
        at away3d.core.raycast::MouseRaycast()
        at away3d.core.managers::Mouse3DManager()
        at away3d.containers::View3D()
        at GameClass/setGlobals()
        at GameClass/retrieveGlobals()
        at Preloader/setGameOn()
        at Preloader/_checkSplashesDone()
```
The new panic should be fixed by #11371.

**Comment by Aaron1011** [comment]
This is working again on `master` (though there's a blue tint caused by `GlowFilter` being unimplemented).
